app/src/main/java/tech/ula/utils/AndroidUtility.kt

```diff
@@ -398,4 +399,54 @@ class AcraWrapper {
     fun putCustomString(key: String, value: String) {
         ACRA.getErrorReporter().putCustomData(key, value)
     }
+}
+
+class UsageUtility(private val prefs: SharedPreferences) {
+    private val numberOfTimesOpenedKey = "numberOfTimesOpened"
+    private val userGaveFeedbackKey = "userGaveFeedback"
+    private val dateTimeFirstOpenKey = "dateTimeFirstOpen"
+    private val millisecondsInThreeDays = 259200000L
+    private val minimumNumberOfOpensBeforeReviewRequest = 15
+
+    fun addToNumberOfTimesOpened(): Int {
+        val numberTimesOpened = prefs.getInt(numberOfTimesOpenedKey, 1)
+
+        if (numberTimesOpened == 1) {
+            with(prefs.edit()) {
+                putInt(numberOfTimesOpenedKey, numberTimesOpened + 1)
+                putLong(dateTimeFirstOpenKey, Date().time)
+                apply()
+            }
+        } else {
+            with(prefs.edit()) {
+                putInt(numberOfTimesOpenedKey, numberTimesOpened + 1)
+                apply()
+            }
+        }
+
+        return numberTimesOpened
+    }
+
+    fun getIsSufficientTimeElapsedSinceFirstOpen(): Boolean {
+        val dateTimeFirstOpened = prefs.getLong(dateTimeFirstOpenKey, 0L)
+        val dateTimeWithSufficientTimeElapsed = dateTimeFirstOpened + millisecondsInThreeDays
+
+        return (Date().time > dateTimeWithSufficientTimeElapsed)
+    }
+
+    fun numberOfTimesOpenedIsGreaterThanThreshold(): Boolean {
+        val numberTimesOpened = prefs.getInt(numberOfTimesOpenedKey, 1)
+        return numberTimesOpened > minimumNumberOfOpensBeforeReviewRequest
+    }
+
+    fun setUserGaveFeedback() {
```

Comment:
What do you think about `UserHasGivenFeedback` instead? Reads a bit cleaner to me

Reply:
That's a great name, I've changed it to `UserHasGivenFeedback`